### PR TITLE
Replace deprecated syntax with curly braces for accessing  array and …

### PR DESCRIPTION
**Pull request by julianhofmann**
…string offset

### CODE REVIEW

File: Classes/Controller/GlossaryController.php [hunk 1]
```diff
@@ -82,7 +82,7 @@ public function listAction(string $letter = ''): void
      */
     public function showAction(Glossary $glossary): void
     {
-        $letter = strtr(mb_strtolower($glossary->getTitle(){0}), 'äöü', 'aou');
+        $letter = strtr(mb_strtolower($glossary->getTitle()[0]), 'äöü', 'aou');
```
Comment (froemken) [inline Classes/Controller/GlossaryController.php:85]: Hi @julianhofmann 

thank you for this PR. As you see I work with mb_strtolower here and I don't know, if [0] supports multi byte, too.
Can you please use mb_substr(title, 0, 1) to get the first letter?

Nice greetings

Stefan

**Review by froemken — CHANGES_REQUESTED** [review]

**Review by froemken — APPROVED** [review]

**Comment by froemken** [comment]
WTF `Unknown database 'test_ft6279efb'`

I have the same issue with maps2 today. Don't get it as it works for TYPO3 9.5 branch?!?!

I will invest some time on friday.

Stefan

**Comment by julianhofmann** [comment]
Strange things with TYPO3 10.4.16? Had today troubles with non initialized ObjectStorages after update 10.4.15 -> 10.4.16. Maybe, you can try it with v10.4.15.

**Comment by froemken** [comment]
Hi @julianhofmann 

I have found the problem:
https://github.com/jweiland-net/glossary2/blob/master/.github/workflows/ci.yml#L57-L59

Can you please update to latest master and build PR again?

Thank you

Stefan

**Comment by froemken** [comment]
Today I have created a new branch for TYPO3 9 called TYPO3_9-5. It contains a versions which is compatible with TYPO3 9 and 10.
In this branch I have just uploaded a patch which make use of mb_substr to get first letter. This version is tagged with 4.3.3.

I will release this version in TER on monday

Have a nice weekend

Stefan

**Pull request closed without merging**